src/grid.rs

```diff
@@ -0,0 +1,269 @@
+use std::thread::sleep;
+use std::time::Duration;
+
+pub struct Grid {
+    matrix: Vec<Vec<u8>>,
+    width: usize,
+    height: usize,
+}
+
+impl Grid {
+    pub fn new(width: usize, height: usize) -> Self {
+        let mut cells: Vec<Vec<u8>> = Vec::with_capacity(width);
+        for _ in 0..width {
+            cells.push(vec![0; height]);
+        }
+
+        Self {
+            matrix: cells,
+            width,
+            height,
+        }
+    }
+
+    pub fn seed(&mut self, points: Vec<(usize, usize)>) {
+        for point in points {
+            self.matrix[point.0][point.1] = 1;
+        }
+    }
+
+    fn count_neighbours(&self, x: usize, y: usize) -> u8 {
+        let mut count = 0;
+        count += if x > 0 {
+            let mut partial_count = 0;
+
+            partial_count += if y > 0 {
+                if self.matrix[x - 1][y - 1] == 1 { 1 } else { 0 }
+            } else {
+                0
+            };
+
+            if self.matrix[x - 1][y] == 1 {
+                partial_count += 1;
+            }
+
+            partial_count += if y < (self.width - 1) {
+                if self.matrix[x - 1][y + 1] == 1 { 1 } else { 0 }
+            } else {
+                0
+            };
+
+            partial_count
+        } else {
+            0
+        };
+
+        count += if y > 0 {
+            let mut partial_count = 0;
+
+            if self.matrix[x][y - 1] == 1 {
+                partial_count += 1;
+            }
+            partial_count += if x < (self.height - 1) {
+                if self.matrix[x + 1][y - 1] == 1 { 1 } else { 0 }
+            } else {
+                0
+            };
+
+            partial_count
+        } else {
+            0
+        };
+
+        count += if x < (self.height - 1) {
+            let mut partial_count = 0;
+
+            partial_count += if self.matrix[x + 1][y] == 1 { 1 } else { 0 };
+
+            partial_count += if y < (self.width - 1) {
+                if self.matrix[x + 1][y + 1] == 1 { 1 } else { 0 }
+            } else {
+                0
+            };
+
+            partial_count
+        } else {
+            0
+        };
+
+        count += if y < (self.width - 1) {
+            if self.matrix[x][y + 1] == 1 { 1 } else { 0 }
+        } else {
+            0
+        };
+
+        count
+    }
+
+    fn clock(&mut self) {
+        let mut new_matrix = self.matrix.clone();
+
+        for (x, row) in new_matrix.iter_mut().enumerate().take(self.width) {
+            for (y, field) in row.iter_mut().enumerate().take(self.width) {
+                let neighbours_ammount = self.count_neighbours(x, y);
```

Comment:
```suggestion
                let neighbours_amount = self.count_neighbours(x, y);
```
The word amount contains only one 'm'